src/lib/axes/axis_utils.ts

```
@@ -598,47 +612,50 @@ export function computeAxisGridLinePositions(
   return positions;
 }
 
-export function getVerticalDomain(
-  xDomain: XDomain,
-  yDomain: YDomain[],
-  chartRotation: number,
-): XDomain | YDomain[] {
-  if (chartRotation === 0 || chartRotation === 180) {
-    return yDomain;
-  } else {
-    return xDomain;
-  }
-}
-
-export function getHorizontalDomain(
-  xDomain: XDomain,
-  yDomain: YDomain[],
-  chartRotation: number,
-): XDomain | YDomain[] {
-  if (chartRotation === 0 || chartRotation === 180) {
-    return xDomain;
-  } else {
-    return yDomain;
-  }
-}
-
-export function getAxisDomain(
-  position: Position,
-  xDomain: XDomain,
-  yDomain: YDomain[],
-  chartRotation: number,
-): XDomain | YDomain[] {
-  if (!isHorizontal(position)) {
-    return getVerticalDomain(xDomain, yDomain, chartRotation);
-  } else {
-    return getHorizontalDomain(xDomain, yDomain, chartRotation);
-  }
-}
-
 export function isVertical(position: Position) {
   return position === Position.Left || position === Position.Right;
 }
 
 export function isHorizontal(position: Position) {
   return !isVertical(position);
 }
+
+export function mergeDomainsByGroupId(
+  axesSpecs: Map<AxisId, AxisSpec>,
+  chartRotation: Rotation,
+): Map<GroupId, DomainRange> {
+  const domainsByGroupId = new Map<GroupId, DomainRange>();
+  axesSpecs.forEach((spec: AxisSpec, id: AxisId) => {
+    const { groupId, domain } = spec;
+
+    if (domain) {
+      const isAxisYDomain = isYDomain(spec.position, chartRotation);
+
+      if (!isAxisYDomain) {
+        // tslint:disable-next-line:no-console
+        console.warn(`[Axis ${id}]: custom domain for xDomain should be defined in Settings`);
```

Do we want to throw here instead of warn?  My thought was that this isn't a fatal error as we can just ignore the config, but wondering if throwing is more clear for the user.

---

We can throw an error here seems a bit more clear and will avoid propagating the wrong configuration in various places

---

addressed in https://github.com/elastic/elastic-charts/pull/98/commits/faf9f25ee155a1c0d05edbf8de40a166ed61da47